src/import/load-costume.js

```diff
@@ -29,9 +31,29 @@ const loadCostumeFromAsset = function (costume, costumeAsset, runtime) {
         ];
     }
     if (costumeAsset.assetType === AssetType.ImageVector) {
+        let svgString = costumeAsset.decodeText();
+        // SVG Renderer load fixes "quirks" associated with Scratch 2 projects
+        // Currently costume.version only exists for scratch 2 costumes
+        if (costume.version && costume.version === 2) {
+            const svgRenderer = new SvgRenderer();
+            svgRenderer.loadString(svgString);
+            svgString = svgRenderer.toString();
+            delete costume.version;
+
+            // Put back into storage
+            const storage = runtime.storage;
+            costume.assetId = storage.builtinHelper.cache(
+                storage.AssetType.ImageVector,
+                storage.DataFormat.SVG,
+                (new TextEncoder()).encode(svgString)
+            );
```

Comment:
Could you use `costumeAsset.encodeTextData(...)` here instead of calling the `TextEncoder` directly? If so you might be able to skip the `TextEncoder` dependency in this repository.

Reply:
Cool! I tried to switch it to use encodeTextData